packages/flutter_tools/lib/src/compile.dart

```diff
@@ -80,11 +82,11 @@ class KernelCompiler {
   Future<CompilerOutput> compile({
     String sdkRoot,
     String mainPath,
-    String outputFilePath,
+    String outputFilePathNoExtension,
     String depFilePath,
     bool linkPlatformKernelIn = false,
     bool aot = false,
-    bool trackWidgetCreation = false,
+    @required bool trackWidgetCreation,
```

Comment:
make this required to defensively avoid bugs where parameters are not passed through as required.
It would be ideal in the future to have some BuildConfig object storing all the parameters we are passing around here so there aren't any bugs flowing the args between the methods but making parameters required is the next best thing.